torchgeo/datamodules/geonrw.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""GeoNRW datamodule."""
+
+import os
+from typing import Any
+
+import kornia.augmentation as K
+from torch.utils.data import Subset
+
+from ..datasets import GeoNRW
+from ..transforms import AugmentationSequential
+from .geo import NonGeoDataModule
+from .utils import group_shuffle_split
+
+
+class GeoNRWDataModule(NonGeoDataModule):
+    """LightningDataModule implementation for the GeoNRW dataset.
+
+    Implements 80/20 train/val splits based on city locations.
+    See :func:`setup` for more details.
```

Comment:
The `setup` function docstring doesn't actually contain more details, not sure if this sentence is helpful.

Reply:
But in the setup function I can see how the train and val indices are generated from a group_shuffle_split and how that is implemented?

Reply:
Yes, but if you're looking at the documentation, this link doesn't help, and if you're looking at the code, you can probably already see the implementation details. For example, see the rendered docs for https://torchgeo.readthedocs.io/en/stable/api/datamodules.html#tropical-cyclone. The link to the setup function isn't very helpful in this case.